src/compiler/checker.ts

```diff
@@ -4632,6 +4659,43 @@ module ts {
                     }
                     result &= related;
                 }
+
+                if (source.typePredicate && target.typePredicate) {
+                    let hasDifferentParamaterIndex = source.typePredicate.parameterIndex !== target.typePredicate.parameterIndex;
+                    let hasDifferentTypes = !isTypeIdenticalTo(source.typePredicate.type, target.typePredicate.type);

```

Comment:
Fixed both problems. I hope you mean by defer by just adding another if statement to 
```typescript
if(hasDifferentParamaterIndex) {?
   let hasDifferentTypes... 
```